hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityRepository.java

```diff
@@ -20,9 +20,31 @@
  * ‍
  */
 
+import static com.hedera.mirror.importer.config.CacheConfiguration.ACCOUNT_ALIAS_CACHE;
+
+import java.util.Optional;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 
 import com.hedera.mirror.common.domain.entity.Entity;
 
 public interface EntityRepository extends CrudRepository<Entity, Long> {
+    @Cacheable(cacheNames = "entityAlias", cacheManager = ACCOUNT_ALIAS_CACHE)
+    @Query(value = "select id from entity where alias = ?1 and deleted <> true", nativeQuery = true)
+    Optional<Long> findByAlias(byte[] alias);
+
+    @Cacheable(cacheNames = "entityAlias", cacheManager = ACCOUNT_ALIAS_CACHE)
+    @Query(value = "select alias from entity where id = ?1 and alias is not null", nativeQuery = true)
+    Optional<byte[]> findByIdWithAliasNotNull(Long id);
+
+    @CacheEvict(cacheNames = "entityAlias", cacheManager = ACCOUNT_ALIAS_CACHE)
+    default void evictAliasCache(byte[] alias) {
+    }
+
+    @CachePut(cacheNames = "entityAlias", cacheManager = ACCOUNT_ALIAS_CACHE)
+    default void addToAliasCache(byte[] alias, Long id) {
+    }
```

Comment:
This method isn't working. `@CachePut` needs to return a value in order to work. I'm also not a fan of the cache-specific name:

```suggestion
    @CachePut(cacheNames = "entityAlias", cacheManager = ACCOUNT_ALIAS_CACHE, key = "#p0")
    default Long storeAlias(byte[] alias, Long id) {
        return id;
    }
```

Reply:
Updated